### DISCUSSION TITLE
Indexing performance

### DISCUSSION THREAD

**Discussion by aliuq** [photoprism/photoprism#1770] [Q&A] [answered]
I have about 100000+ pictures, photoprism running on Synology 920(2vCPU 4g), Is this speed correct？

![image](https://user-images.githubusercontent.com/11707934/143792692-3ca83ba2-e0b7-4e93-98cd-db9a07276f3e.png)

**Comment by lastzero** [comment, marked as answer]
When you have large tables (many files) it may help to increase the InnoDB cache size as described in the MariaDB documentation. Also make sure Docker didn't put the database files on a slow / remote disk for whatever reason.

See https://github.com/photoprism/photoprism/discussions/1768#discussioncomment-1729552 for the general performance of servers based on Intel Atom / Celeron CPUs. They may be comparatively fast for a NAS, but not very fast in absolute terms. You shouldn't expect a wonder.

**Reply by aliuq** [reply]
Thank you

**Reply by lastzero** [reply]
Added a section on MariaDB Performance Tuning to our docs: https://docs.photoprism.org/getting-started/troubleshooting/#performance-tuning
